### DISCUSSION TITLE
assign a workspace to a given monitor doesn't work anymore.

### DISCUSSION THREAD

**Discussion by ethanfel** [regolith-linux/regolith-desktop#718] [Q&A] [answered]
Hi ! 
I've been using regolith for years now and i've always been using workspace assignement. I just updated to Regolith 2.

i did stage my config following [this](https://regolith-desktop.com/docs/howtos/stage-configs/)

and added the code i was using before with 1.6 (it follow the [wiki](https://regolith-desktop.com/docs/howtos/workspace-on-output/)
but it's not working anymore (lots of error after logout)

can you help me finding the correct way to do this with regolith 2 ? thanks

**Comment by ethanfel** [comment]
i've made some progress. (it actually work now with some errors)

Xressource files didn't work.

So i've started using configs partials. Followed the wiki to create my own 40_workspace-config and remove the original one with apt.

added [my settings](https://imgur.com/a/hK9NOc9) and after a refresh it work.

But it doesn't work after a reboot/logout (the errors are in the previous link). A refresh is needed to fix it.

**Comment by jschwuch** [comment, marked as answer]
Hi,

I'm a new user to regolith but am migrating from an existing i3 config and had a similar issue. I just want to share the solution that works for me and give some info about what I tried and didn't work.

### TLDR
Remove the magic from the workspace names. Like: `i3-wm.workspace.01.name: 1`

### First Try: Using my old config

First I tried to use my old i3 config. I stripped everything that I already configured via Xresources and was basically left with 

```
workspace 1 output $leftScreen
workspace 2 output $leftScreen
workspace 3 output $leftScreen
workspace 4 output $rightScreen
[...]

assign [title=".*Google Chrome$"] 3
assign [class="Microsoft Teams - Preview"] 1
[...]
```
plus a few additional key bindings. I put this into `~/.config/regolith2/i3/config.d/100_user_config`.

This kind of worked but duplicated the workspaces. I had all my old workspaces where I wanted them and additionally all the workspaces with a colored circle after the name. My applications were moved to the workspaces without the circles.


### Second try
To move my applications to the correct workspaces (those with circles) I added the `number` keyword to the `assign` commands. This worked as expected.

But assigning the workspaces to the correct displays didn't work properly, I got the same error messages as @ethanfel.

```
workspace "$ws1" output $leftScreen
workspace "$ws2" output $leftScreen
workspace "$ws3" output $leftScreen
workspace "$ws4" output $rightScreen
[...]

assign [title=".*Google Chrome$"] number 3
assign [class="Microsoft Teams - Preview"] number 1
[...]
```

It looks like i3 can't properly process the HTML polluted workspace names so I got rid of them in

### The final try

I set the workspace names of all the workspaces 1-10 to only numbers via the `~/.config/regolith2/Xresources` file like this:

```
i3-wm.workspace.01.name: 1
i3-wm.workspace.02.name: 2
i3-wm.workspace.03.name: 3
i3-wm.workspace.04.name: 4
i3-wm.workspace.05.name: 5
i3-wm.workspace.06.name: 6
i3-wm.workspace.07.name: 7
i3-wm.workspace.08.name: 8
i3-wm.workspace.09.name: 9
i3-wm.workspace.10.name: 10
```

and changed my `100_user_config` back to the state from the first try and everything works fine!

I hope this might help somebody who struggles with the same problem. Ideally, this would work out of the box and it should be documented how to achieve this.

**Comment by ethanfel** [comment]
ho nice, thanks for the detailed answer.

did you add the assign lines to the Xressources file to ?

**Reply by jschwuch** [reply]
No, both, the lines starting with `workspace` and the lines starting with `assign` need to go into your config partial. In my case that's `~/.config/regolith2/i3/config.d/100_user_config`

**Reply by ethanfel** [reply]
it's working perfectly like before ! thank you very much.
